course/src/isometry.cpp

```diff
@@ -3,15 +3,176 @@
  * Author: Agustin Alba Chicar, 2019
  * Author: Gerardo Puga, 2020
  */
+// Copyright 2020, Blast545
 
 #include <isometry/isometry.hpp>
 
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+  Vector3::Vector3(double x, double y, double z) :
+    x_(x), y_(y), z_(z) {}
+
+  Vector3::Vector3() :
+    x_(1.0), y_(2.0), z_(3.0) {}
+
+  double Vector3::norm() const {
+    return sqrt(pow(x_, 2) + pow(y_, 2) + pow(z_, 2));
+  }
+
+  double Vector3::x() const {
+    return x_;
+  }
+
+  double & Vector3::x() {
+    return x_;
+  }
+
+  double Vector3::y() const {
+    return y_;
+  }
+
+  double & Vector3::y() {
+    return y_;
+  }
+
+  double Vector3::z() const {
+    return z_;
+  }
+
+  double & Vector3::z() {
+    return z_;
+  }
+
+  double Vector3::dot(const Vector3& vector1) const {
+    return x_*vector1.x() + y_*vector1.y() + z_*vector1.z();
+  }
+
+  Vector3 Vector3::cross(const Vector3& vector1) const {
+    return {
+            y_*vector1.z() - z_*vector1.y(),
+            z_*vector1.x() - x_*vector1.z(),
+            x_*vector1.y() - y_*vector1.x()};
+  }
+
+  const bool Vector3::operator==(const Vector3& vector1) const {
+    return(
+      (x_ == vector1.x_) &&
```

Comment:
Remember that there are instructions in the challenge about how to go about testing equality of floating point variables: https://floating-point-gui.de/errors/comparison/#:~:text=Due%20to%20rounding%20errors%2C%20most,end%20up%20being%20slightly%20imprecise.&text=However%2C%20it%20also%20means%20that,a%20simple%20equality%20test%20fails.